eo-maven-plugin/src/main/java/org/eolang/maven/ResolveMojo.java

```diff
@@ -218,70 +213,53 @@ private Collection<Dependency> deps() throws IOException {
     }
 
     /**
-     * Check if all dependencies have transitive dependencies.
+     * Dependencies object.
      *
-     * @param deps Dependencies
-     * @throws java.lang.IllegalStateException if a transitive dependency is found
+     * @param all Dependencies as a collection
+     * @return Dependencies object with applied decorators.
      */
-    private void checkTransitive(final Collection<Dependency> deps) {
-        if (!this.ignoreTransitive) {
-            for (final Dependency dep : deps) {
-                if (new Unchecked<>(
-                    new LengthOf(
-                        new DcsTransitive(
-                            new DcsDepgraph(
-                                project,
-                                session,
-                                manager,
-                                this.targetDir.toPath()
-                                    .resolve(ResolveMojo.DIR)
-                                    .resolve("dependencies-info"),
-                                dep
-                            ),
-                            dep
-                        ))).value() != 0) {
-                    throw new IllegalStateException(
-                        String.format("%s contains transitive dependencies", dep)
-                    );
-                }
-            }
-        }
-    }
-
-    /**
-     * Check dependencies for conflicts.
-     *
-     * @param deps Dependencies
-     */
-    private void checkConflicts(final Collection<Dependency> deps) {
-        final Map<String, Set<String>> conflicts = deps.stream()
-            .collect(
-                Collectors.groupingBy(
-                    Dependency::getManagementKey,
-                    Collectors.mapping(
-                        Dependency::getVersion,
-                        Collectors.toSet()
-                    )
-                )
-            ).entrySet().stream()
-            .filter(e -> e.getValue().size() > 1)
-            .collect(
-                Collectors.toMap(
-                    Map.Entry::getKey,
-                    Map.Entry::getValue
+    private Dependencies dependenciesOf(final Collection<Dependency> all) {
+        final Dependencies dependencies;
+        if (this.ignoreTransitive && this.ignoreVersionConflicts) {
+            dependencies = new DcsWithRuntime(all::iterator);
```

Comment:
@volodya-lombrozo looks like code duplication to me. You use `DcsWithRuntime` everywhere in each `if`. Maybe it's better to make `dependencies` mutable?

Reply:
@yegor256 done.